### PR TITLE
bug(nimbus): Fix the nimbus-rust-image container

### LINKED ISSUE #9707: Errors building nimbus-rust-image (Dockerfile-rust-image)
When #9694 landed, it broke building the nimbus-rust-image container:

```
View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/zf3u6mh8hrft2l6ps0p7v8iqo
docker buildx build -t experimenter:nimbus-rust-image -f experimenter/tests/integration/nimbus/utils/Dockerfile-ruts-image
ERROR: "docker buildx build" requires exactly 1 argument.
See 'docker buildx build --help'.

Usage:  docker buildx build [OPTIONS] PATH | URL | -

Start a build
make: *** [build_rust_image] Error 1
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4014)

### PR DESCRIPTION
**Pull request by brennie**
Because:

- we broke the building of the nimbus-rust-image container;
- we now have the nimbus-experimenter megazord in the experimenter
  containers;
- the container broke without CI failing the PR

This commit:

- removes the custom `Dockerfile-rust-image` and adds a stage to the
  experimenter Dockerfile;
- adds a new task to build (but not deploy) the nimbus-rust-image
  container in PRs; and
- adds a new task to deploy the nimbus-rust-image container on updates
  to main.

Fixes #9707

### CODE REVIEW

**Review by jaredlockhart — COMMENTED** [review]
Any changes needed in circle config?

**Comment by jrbenny35** [comment]
You might need to change this to point to the new image https://github.com/mozilla/experimenter/blob/main/experimenter/tests/integration/tox.ini#L25

File: experimenter/Dockerfile [hunk 1]
```diff
@@ -127,6 +127,18 @@ COPY --from=file-loader /experimenter/experimenter/templates/ /experimenter/expe
...
+#-------------------------
+FROM python:3.11.1 AS nimbus-rust-image
+WORKDIR /code
+
+RUN apt-get update -qqy && \
+    apt-get install -qqy \
+        tox
+
+COPY --from=dev /application-services /application-services/
+ENV PYTHONPATH=$PYTHONPATH:/application-services
```
Comment (jrbenny35) [inline experimenter/Dockerfile:140]: Is this the path of the ```nimbus.py``` file?

**Comment by brennie** [comment]
Closed in favour of #9712

**Pull request closed without merging**